### ISSUE TITLE
Incoherent elements given to the model in Insertion/Deletion metrics

### ISSUE THREAD

**Issue by AntoninPoche** [FOR-sight-ai/interpreto#110]
## 🐛 Bug Report

When calling the metrics on explanations from gradient-based methods (`IntegratedGradients`, `SmoothGrad`...), it crashes during the model call.

Incoherent input_ids and attention_mask are provided to the model.

I expect that the `model_inputs_to_explain` loaded from the `AttentionOutput` are `inputs_embeds` with gradient-based methods, and should not be.

## 🔬 How To Reproduce

Interpreto 0.4.8

## Expected behavior

Metrics work for all methods.

## Additional context

This might be linked in some sens to #109
